collection.go

```diff
@@ -23,38 +31,85 @@ type IndexingPolicy struct {
 }
 
 type IndexingMode string
-
-const (
-	Consistent = IndexingMode("Consistent")
-	Lazy       = IndexingMode("Lazy")
-)
+type OfferThroughput int32
+type OfferType string
+
+//const (
+//	Consistent = IndexingMode("Consistent")
+//	Lazy       = IndexingMode("Lazy")
+//)
+//
+//const (
+//	OfferTypeS1 = OfferType("S1")
+//	OfferTypeS2 = OfferType("S2")
+//	OfferTypeS3 = OfferType("S3")
+//)
 
 type PartitionKey struct {
 	Paths []string `json:"paths"`
 	Kind  string   `json:"kind"`
 }
 
+// https://docs.microsoft.com/en-us/rest/api/cosmos-db/create-a-collection
```

Comment:
I think we should rename things a bit. Everythin with the `...Options` suffix are _headers_ so far. E.g. `document.go`, the `CreateDocumentOptions` struct contains all required and optional headers. I think we should keep a consistent naming.

Reply:
On the other hand, since the payload in the expected body is fixed and known I see that we could bundle it and in a way the caller should not need to have knowledge wether the options have to be provided as header or in the body.